src/tools/clippy/clippy_lints/src/loops/needless_range_loop.rs

```diff
@@ -57,7 +57,13 @@ pub(super) fn check<'tcx>(
                 if let Some(indexed_extent) = indexed_extent {
                     let parent_def_id = cx.tcx.hir().get_parent_item(expr.hir_id);
                     let region_scope_tree = cx.tcx.region_scope_tree(parent_def_id);
-                    let pat_extent = region_scope_tree.var_scope(pat.hir_id.local_id).unwrap();
+                    let var_scope = if cx.tcx.sess.at_least_rust_2024() && cx.tcx.features().new_temp_lifetime {
+                        let scope_map = cx.tcx.body_scope_map(parent_def_id);
+                        scope_map.var_scope_new(pat.hir_id.local_id)
+                    } else {
+                        region_scope_tree.var_scope(pat.hir_id.local_id)
+                    };
```

Comment:
I have given this a thought. The idea is that I would like to keep the `body_scope_map` behind the feature gate and leave the original scope tree implementation untouched so that both can evolve independently.